src/HookDispatcher.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * 2007-2019 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShop\Module\FacetedSearch;
+
+use Ps_Facetedsearch;
+use PrestaShop\Module\FacetedSearch\Hook;
+
+class HookDispatcher
```

Comment:
Hook dispatcher is here to dispatch hook, I think it's so obvious and don't need this kind of useless comment :/

Reply:
I just of a comment to explain that explains that this class works with `AbstractHook` instances in order to reduce the module size, and that it is based on the `__call` method from the module.
Ok `HookDispatcher` is self explanatory but as there is already one in the core it can help to know what this specific one does in this module context.

Reply:
No problem! :smiley: